### ISSUE TITLE
b-table sticky header broken if sortable

### ISSUE THREAD

**Issue by sebastian-raubach** [bootstrap-vue/bootstrap-vue#6961]
### Describe the bug

When using `b-table` with the `sticky-header` prop and sortable columns, the stickyness of those columns is broken. 

Changing the example on https://bootstrap-vue.org/docs/components/table#sticky-headers to this, shows the problem:

```javascript
<template>
  <div>
    <b-table sticky-header :fields="columns" :items="items" head-variant="light"></b-table>
  </div>
</template>

<script>
  export default {
    data() {
      return {
        columns: [{ key: 'heading1', sortable: true }, { key: 'heading2', sortable: false }],
        items: [
          { heading1: 'table cell', heading2: 'table cell', heading3: 'table cell' },
          { heading1: 'table cell', heading2: 'table cell', heading3: 'table cell' },
          { heading1: 'table cell', heading2: 'table cell', heading3: 'table cell' },
          { heading1: 'table cell', heading2: 'table cell', heading3: 'table cell' },
          { heading1: 'table cell', heading2: 'table cell', heading3: 'table cell' },
          { heading1: 'table cell', heading2: 'table cell', heading3: 'table cell' },
          { heading1: 'table cell', heading2: 'table cell', heading3: 'table cell' },
          { heading1: 'table cell', heading2: 'table cell', heading3: 'table cell' },
          { heading1: 'table cell', heading2: 'table cell', heading3: 'table cell' },
          { heading1: 'table cell', heading2: 'table cell', heading3: 'table cell' },
          { heading1: 'table cell', heading2: 'table cell', heading3: 'table cell' },
          { heading1: 'table cell', heading2: 'table cell', heading3: 'table cell' }
        ]
      }
    }
  }
</script>
```

### Steps to reproduce the bug

1. Go to https://bootstrap-vue.org/docs/components/table#sticky-headers
2. Double click the live code section
3. Paste in above code
4. Scroll table down to see only one header being sticky

### Expected behavior

All headers should be sticky.

### Versions

**Libraries:**

- BootstrapVue: 2.22.0
- Bootstrap: 4.6.1
- Vue: 2.6.14

**Environment:**

- Device: Any
- OS: Windows, Android
- Browser: Chrome
- Version: 100.0.4896.127

### Demo link

See description above.

**Comment by 2000ice** [comment]
Both sticky header and sticky column are broken.
Re-produce:
Go to https://bootstrap-vue.org/docs/components/table#sticky-columns
Add "sortable:true" in one field, like
{ key: 'id', stickyColumn: true, isRowHeader: true, variant: 'primary', sortable:true},

Header and sticky column header (only the header with sortable) will disappear, if scrolling horizontally or vertically.

**Comment by driehle** [comment]
It looks liks PR #6371 introduced the bug in 2.22.0 with its change in `src/components/table/helpers/mixin-thead.js`:

```diff
diff --git a/src/components/table/helpers/mixin-thead.js b/src/components/table/helpers/mixin-thead.js
index a767de0fa78..f5126f8ea61 100644
--- a/src/components/table/helpers/mixin-thead.js
+++ b/src/components/table/helpers/mixin-thead.js
@@ -113,7 +113,15 @@ export const theadMixin = Vue.extend({
         const sortLabel = isSortable ? this.sortTheadThLabel(key, field, isFoot) : null
 
         const data = {
-          class: [this.fieldClasses(field), sortClass],
+          class: [
+            {
+              // We need to make the header cell relative when we have
+              // a `.sr-only` sort label to work around overflow issues
+              'position-relative': sortLabel
+            },
+            this.fieldClasses(field),
+            sortClass
+          ],
           props: { variant, stickyColumn },
           style: field.thStyle || {},
           attrs: {
```

**Comment by luca-albani** [comment]
I noticed that when the table is set with sticky-header and sortable column, this is the resulting markup of the sortable th:
`<th role="columnheader" scope="col" tabindex="0" aria-colindex="4" aria-sort="none" class="table-info b-table-sticky-column position-relative"><div class="text-nowrap">
          Heading C
        </div><span class="sr-only"> (Click to sort ascending)</span></th>`

        It looks like the issue depends on class "position-relative" set on the th element.
        Try setting position-sticky solves the issue.

**Comment by m-mohr** [comment]
I have the same issue. I simply fixed it for me by adding the following to my local CSS:

```css
/*
  Fix an issue in vue-bootstrap v2.22.0:
  https://github.com/bootstrap-vue/bootstrap-vue/issues/6961 */
.b-table-sticky-header > .table.b-table > thead > tr > th {
  position: sticky !important;
}
```

And I assume the issue could be fixed here by simply changing `'position-relative': sortLabel` to `'position-relative': sortLabel && !stickyColumn`?!

**Comment by JackSlaterIV** [comment]
Hello, this issue is still present on v.2.23.1.
I saw @m-mohr sent a PR which was merged, but actually it only fixes the problem for the columns not sticking.
The table header still does not stick if the columns are set as sortable.
Please anyone fix this issue.
